enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/mtls/MTLSAuthenticator.java

```diff
@@ -0,0 +1,190 @@
+                            "the API: %s", requestContext.getRequestPath(), requestContext.getMatchedAPI().getName()));
+                    clientCert = null;
+                }
+                if (clientCert != null) {
+                    authenticated = true;
+                    clientCertificateTier = MtlsUtils.getTier(clientCertificateAlias, mtlsCertificateTiers);
+                    if (clientCertificateTier != null) {
+                        if (!clientCertificateTier.equals("")) {
+                            authenticationContext.setTier(clientCertificateTier);
+                        }
+                    }
+                    String subjectDN = clientCert.getSubjectDN().getName();
+                    authenticationContext.setUsername(subjectDN);
+                }
+            } catch (CertificateException e) {
+                throw new APISecurityException(APIConstants.StatusCodes.UNAUTHENTICATED.getCode(),
+                        APISecurityConstants.API_AUTH_GENERAL_ERROR, "Invalid client certificate");
+            }
+
+            String apiName = requestContext.getMatchedAPI().getName();
+            String apiVersion = requestContext.getMatchedAPI().getVersion();
+            String apiUUID = requestContext.getMatchedAPI().getUuid();
+
+            authenticationContext.setAuthenticated(authenticated);
+            authenticationContext.setApiName(apiName);
+            authenticationContext.setApiUUID(apiUUID);
+            authenticationContext.setApiVersion(apiVersion);
+
+            return authenticationContext;
+        } finally {
+            if (Utils.tracingEnabled()) {
+                mtlsApiAuthenticatorSpanScope.close();
+                Utils.finishSpan(mtlsApiAuthenticatorSpan);
+            }
+        }
+    }
+
+    private X509Certificate getClientCertificate(RequestContext requestContext) throws CertificateException {
+        String certContent = "";
+
+        if (isEnableClientValidation) {
+            String encodedCert = requestContext.getClientCertificate();
+            if (!encodedCert.isEmpty()) {
+                certContent = MtlsUtils.getCertContent(encodedCert, true);
+            }
+        } else {
+            Map<String, String> headers = requestContext.getHeaders();
+            if (headers.containsKey(FilterUtils.getCertificateHeaderName())) {
+                String cert = retrieveMTLSHeaderValue(requestContext);
+                requestContext.setClientCertificate(cert);
+                if (!cert.isEmpty()) {
+                    certContent = MtlsUtils.getCertContent(cert, isClientCertificateEncode);
+                }
+            }
+        }
+        if (!certContent.isEmpty()) {
+            return MtlsUtils.getX509Cert(certContent);
+        }
+        log.debug(String.format("Provided client certificate in the request: %s is invalid",
+                requestContext.getRequestPath()));
+        return null;
+    }
+
```

Comment:
Why do we require methods like retrieveMTLSHeaderValue, retrieveTrustStore, retrieveMtlsCertificateTiers. as far as i can see those methods simply returns content already in  request context without any further processings

Reply:
Yes. They are not doing any processing. I will remove those methods.